Can't configure CVC5 with `--cocoa --gpl` when building on Windows

**Describe the bug**

On Windows, if you configure CVC5 with the `--cocoa --gpl` options and then attempt to build it, it will fail when building CoCoALib.

**Command line arguments**: Command line arguments passed to cvc5 if applicable  
**cvc5 version/commit**: CVC5 1.3.1 (ea1b484fa54bfe56c0f8b3ac90a6e3e2f46441e7)
**Operating system**: Windows 10 / Microsoft Windows Server 2022 (10.0.20348)

**`configure.sh` options**  

```
./configure.sh --static --static-binary --auto-download --cocoa --gpl --win64-native production
```

**`configure.sh` output**  

<details>

```
-- Building for: MSYS Makefiles
-- The C compiler identification is GNU 16.1.0
-- The CXX compiler identification is GNU 16.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: D:/a/_temp/msys64/mingw64/bin/cc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: D:/a/_temp/msys64/mingw64/bin/c++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: D:/a/_temp/msys64/usr/bin/git.exe (found version "2.54.0")
-- Building Production build
-- Performing Test HAVE_C_FLAG_O3
-- Performing Test HAVE_C_FLAG_O3 - Success
-- Configuring with C flag '-O3'
-- Performing Test HAVE_CXX_FLAG_O3
-- Performing Test HAVE_CXX_FLAG_O3 - Success
-- Configuring with CXX flag '-O3'
-- Performing Test HAVE_C_FLAG_Wall
-- Performing Test HAVE_C_FLAG_Wall - Success
-- Configuring with C flag '-Wall'
-- Performing Test HAVE_CXX_FLAG_Wall
-- Performing Test HAVE_CXX_FLAG_Wall - Success
-- Configuring with CXX flag '-Wall'
-- Performing Test HAVE_C_FLAG_Wunused_private_field
-- Performing Test HAVE_C_FLAG_Wunused_private_field - Failed
-- Performing Test HAVE_CXX_FLAG_Wunused_private_field
-- Performing Test HAVE_CXX_FLAG_Wunused_private_field - Failed
-- Performing Test HAVE_C_FLAG_Wdangling_reference
-- Performing Test HAVE_C_FLAG_Wdangling_reference - Failed
-- Performing Test HAVE_CXX_FLAG_Wdangling_reference
-- Performing Test HAVE_CXX_FLAG_Wdangling_reference - Success
-- Configuring with CXX flag '-Wno-dangling-reference'
-- Performing Test HAVE_C_FLAG_fexceptions
-- Performing Test HAVE_C_FLAG_fexceptions - Success
-- Configuring with C flag '-fexceptions'
-- Performing Test HAVE_CXX_FLAG_Wsuggest_override
-- Performing Test HAVE_CXX_FLAG_Wsuggest_override - Success
-- Configuring with CXX flag '-Wsuggest-override'
-- Performing Test HAVE_CXX_FLAG_Wnon_virtual_dtor
-- Performing Test HAVE_CXX_FLAG_Wnon_virtual_dtor - Success
-- Configuring with CXX flag '-Wnon-virtual-dtor'
-- Performing Test HAVE_C_FLAG_Wimplicit_fallthrough
-- Performing Test HAVE_C_FLAG_Wimplicit_fallthrough - Success
-- Configuring with C flag '-Wimplicit-fallthrough'
-- Performing Test HAVE_CXX_FLAG_Wimplicit_fallthrough
-- Performing Test HAVE_CXX_FLAG_Wimplicit_fallthrough - Success
-- Configuring with CXX flag '-Wimplicit-fallthrough'
-- Performing Test HAVE_C_FLAG_Wshadow
-- Performing Test HAVE_C_FLAG_Wshadow - Success
-- Configuring with C flag '-Wshadow'
-- Performing Test HAVE_CXX_FLAG_Wshadow
-- Performing Test HAVE_CXX_FLAG_Wshadow - Success
-- Configuring with CXX flag '-Wshadow'
-- Performing Test HAVE_C_FLAG_fno_operator_names
-- Performing Test HAVE_C_FLAG_fno_operator_names - Failed
-- Performing Test HAVE_CXX_FLAG_fno_operator_names
-- Performing Test HAVE_CXX_FLAG_fno_operator_names - Success
-- Configuring with CXX flag '-fno-operator-names'
-- Performing Test HAVE_CXX_FLAG_fno_extern_tls_init
-- Performing Test HAVE_CXX_FLAG_fno_extern_tls_init - Success
-- Configuring with CXX flag '-fno-extern-tls-init'
-- Performing Test HAVE_CXX_FLAG_Wclass_memaccess
-- Performing Test HAVE_CXX_FLAG_Wclass_memaccess - Success
-- Configuring with CXX flag '-Wno-class-memaccess'
-- Disabling unit tests since assertions are disabled.
-- Found Python: C:\hostedtoolcache\windows\Python\3.10.11\x64/python.exe (found version "3.10.11") found components: Interpreter
-- Creating Python virtual environment
-- Found Python: D:/a/what4-solvers/what4-solvers/repos/cvc5/build/venv-3.10.11/Scripts/python.exe (found version "3.10.11") found components: Interpreter
-- Found shell interpreter: D:/a/_temp/msys64/usr/bin/sh.exe
-- Found GMP (unknown version): D:/a/_temp/msys64/mingw64/lib/libgmp.a
-- Found shell interpreter: D:/a/_temp/msys64/usr/bin/sh.exe
-- Looking for getc_unlocked
-- Looking for getc_unlocked - not found
-- Looking for closefrom
-- Looking for closefrom - not found
-- Building CaDiCaL rel-2.1.3-elevate: D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/lib/libcadical.a
-- Found shell interpreter: D:/a/_temp/msys64/usr/bin/sh.exe
-- Building Poly 0.2.0: D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/lib/libpicpoly.a, D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/lib/libpicpolyxx.a
-- Found shell interpreter: D:/a/_temp/msys64/usr/bin/sh.exe
-- Building CoCoA 0.99800: D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/lib/libcocoa.a
-- Found shell interpreter: D:/a/_temp/msys64/usr/bin/sh.exe
-- Building SymFPU: D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/include/
-- Performing Test CVC5_NEED_INT64_T_OVERLOADS
-- Performing Test CVC5_NEED_INT64_T_OVERLOADS - Success
-- Performing Test CVC5_NEED_HASH_UINT64_T_OVERLOAD
-- Performing Test CVC5_NEED_HASH_UINT64_T_OVERLOAD - Failed
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for sys/wait.h
-- Looking for sys/wait.h - not found
-- Looking for C++ include ext/stdio_filebuf.h
-- Looking for C++ include ext/stdio_filebuf.h - found
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Configuring with C flag '-pthread'
-- Configuring with CXX flag '-pthread'
-- Looking for ffs
-- Looking for ffs - not found
-- Looking for optreset
-- Looking for optreset - not found
-- Looking for sigaltstack
-- Looking for sigaltstack - not found
-- Looking for strerror_r
-- Looking for strerror_r - not found
-- Looking for strtok_r
-- Looking for strtok_r - found
-- Looking for setitimer
-- Looking for setitimer - not found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Performing Test STRERROR_R_CHAR_P
-- Performing Test STRERROR_R_CHAR_P - Failed
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success

cvc5 1.3.1-modified

Build profile             : production

Assertions                : off
Debug symbols             : off
Debug context mem mgr     : off

Muzzle                    : off
Statistics                : on
Tracing                   : off

ASan                      : off
UBSan                     : off
TSan                      : off
Coverage (gcov)           : off
Profiling (gprof)         : off
Unit tests                : off
Valgrind                  : off

Shared build              : off
Python bindings           : off
Java bindings             : off
Interprocedural opt.      : off

CryptoMiniSat             : off
GLPK                      : off
Kissat                    : off
LibPoly                   : on (local)
CoCoALib                  : on (local)
MP library                : gmp (system)
Editline                  : off

Api docs                  : off


CPPLAGS (-D...): NDEBUG CVC5_STATISTICS_ON CVC5_USE_POLY CVC5_USE_COCOA
CXXFLAGS       : -O3 -Wall -Wno-dangling-reference -Wsuggest-override -Wnon-virtual-dtor -Wimplicit-fallthrough -Wshadow -fno-operator-names -fno-extern-tls-init -Wno-class-memaccess -pthread
CFLAGS         : -O3 -Wall -fexceptions -Wimplicit-fallthrough -Wshadow -pthread
Linker flags   :  -Wl,--stack,100000000

Install prefix : C:/Program Files (x86)/cvc5

cvc5 license: GPLv3 (due to optional libraries; see below)

Please note that cvc5 will be built against the following GPLed libraries:
 cocoa
As these libraries are covered under the GPLv3, so is this build of cvc5.
cvc5 is also available to you under the terms of the (modified) BSD license.
If you prefer to license cvc5 under those terms, please configure cvc5 to
disable all optional GPLed library dependencies.

Now change to 'build' and type 'make', followed by 'make check' or 'make install'.

-- Configuring done (20.2s)
-- Generating done (1.0s)
-- Build files have been written to: D:/a/what4-solvers/what4-solvers/repos/cvc5/build
```
</details>

**Build output**

<details>

```
+ cd build
+ make -j4
[  0%] Creating directories for 'CaDiCaL-EP'
[  0%] Creating directories for 'CoCoA-EP'
[  0%] Creating directories for 'SymFPU-EP'
[  0%] Creating directories for 'Poly-EP'
[  0%] Performing download step (download, verify and extract) for 'CoCoA-EP'
[  0%] Performing download step (download, verify and extract) for 'SymFPU-EP'
[  0%] Performing download step (download, verify and extract) for 'CaDiCaL-EP'
[  0%] Performing download step (download, verify and extract) for 'Poly-EP'
-- SymFPU-EP download command succeeded.  See also D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/SymFPU-EP-stamp/SymFPU-EP-download-*.log
-- Poly-EP download command succeeded.  See also D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/Poly-EP-stamp/Poly-EP-download-*.log
[  0%] No update step for 'SymFPU-EP'
-- CaDiCaL-EP download command succeeded.  See also D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/CaDiCaL-EP-stamp/CaDiCaL-EP-download-*.log
[  0%] No update step for 'Poly-EP'
[  1%] No patch step for 'SymFPU-EP'
[  1%] No update step for 'CaDiCaL-EP'
-- CoCoA-EP download command succeeded.  See also D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/CoCoA-EP-stamp/CoCoA-EP-download-*.log
[  1%] No configure step for 'SymFPU-EP'
[  1%] Performing patch step for 'Poly-EP'
[  1%] No patch step for 'CaDiCaL-EP'
-- Poly-EP patch command succeeded.  See also D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/Poly-EP-stamp/Poly-EP-patch-*.log
[  1%] No update step for 'CoCoA-EP'
[  1%] No build step for 'SymFPU-EP'
[  1%] Performing configure step for 'CaDiCaL-EP'
[  1%] Performing configure step for 'Poly-EP'
[  1%] Performing patch step for 'CoCoA-EP'
-- CaDiCaL-EP configure command succeeded.  See also D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/CaDiCaL-EP-stamp/CaDiCaL-EP-configure-*.log
[  1%] Performing install step for 'SymFPU-EP'
-- CoCoA-EP patch command succeeded.  See also D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/CoCoA-EP-stamp/CoCoA-EP-patch-*.log
[  1%] Performing build step for 'CaDiCaL-EP'
[  1%] Performing configure step for 'CoCoA-EP'
-- SymFPU-EP install command succeeded.  See also D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/SymFPU-EP-stamp/SymFPU-EP-install-*.log
[  1%] Completed 'SymFPU-EP'
[  1%] Built target SymFPU-EP
[  1%] Generating type_enumerator.cpp
[  1%] Generating theory_traits.h
[  1%] Generating rewriter_tables.h
[  1%] Built target gen-theory
-- Found Git: D:/a/_temp/msys64/usr/bin/git.exe (found version "2.54.0")
[  1%] Built target gen-versioninfo
[  1%] Generating options/options.stamp
-- Poly-EP configure command succeeded.  See also D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/Poly-EP-stamp/Poly-EP-configure-*.log
[  1%] Built target gen-options
[  1%] Performing build step for 'Poly-EP'
[  1%] Generating Trace_tags.h
[  1%] Built target gen-tags
[  1%] Building CXX object src/context/CMakeFiles/cvc5context.dir/context.cpp.obj
D:/a/what4-solvers/what4-solvers/repos/cvc5/src/context/context.cpp: In function 'std::ostream& cvc5::context::operator<<(std::ostream&, const Context&)':
D:/a/what4-solvers/what4-solvers/repos/cvc5/src/context/context.cpp:283:12: warning: variable 'level' set but not used [-Wunused-but-set-variable=]
  283 |   uint32_t level = context.d_scopeList.size() - 1;
      |            ^~~~~
[  2%] Building CXX object src/context/CMakeFiles/cvc5context.dir/context_mm.cpp.obj
[  2%] Built target cvc5context
[  2%] Generating kind.h
[  2%] Generating metakind.h
[  2%] Generating node_manager.h
[  2%] Generating type_checker.cpp
[  2%] Generating type_properties.h
[  3%] Generating kind.cpp
[  3%] Generating metakind.cpp
[  3%] Generating node_manager.cpp
[  3%] Generating type_properties.cpp
[  3%] Built target gen-expr
[  4%] Generating rewrites.{h,cpp}
-- CoCoA-EP configure command succeeded.  See also D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/CoCoA-EP-stamp/CoCoA-EP-configure-*.log
[  4%] Performing build step for 'CoCoA-EP'
[  4%] Built target gen-rewrites
[  4%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/commands.cpp.obj
[  4%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/command_status.cpp.obj
[  4%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/input.cpp.obj
[  4%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/lexer.cpp.obj
[  4%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/parser.cpp.obj
[  4%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/parse_op.cpp.obj
[  4%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/parser_state.cpp.obj
[  4%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/parser_utils.cpp.obj
[  5%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/smt2/smt2_state.cpp.obj
[  5%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/smt2/smt2_cmd_parser.cpp.obj
[  5%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/smt2/smt2_lexer.cpp.obj
-- Poly-EP build command succeeded.  See also D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/Poly-EP-stamp/Poly-EP-build-*.log
[  7%] Performing install step for 'Poly-EP'
[  7%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/smt2/smt2_parser.cpp.obj
-- Poly-EP install command succeeded.  See also D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/Poly-EP-stamp/Poly-EP-install-*.log
[  7%] Performing cleanup step for 'Poly-EP'
[  7%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/smt2/smt2_term_parser.cpp.obj
[  7%] Completed 'Poly-EP'
[  7%] Built target Poly-EP
[  7%] Generating smt2_tokens.h
[  7%] Built target gen-tokens
[  7%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/sym_manager.cpp.obj
[  7%] Building CXX object src/base/CMakeFiles/cvc5base.dir/check.cpp.obj
[  7%] Building CXX object src/base/CMakeFiles/cvc5base.dir/configuration.cpp.obj
[  7%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/symbol_table.cpp.obj
[  7%] Building CXX object src/base/CMakeFiles/cvc5base.dir/exception.cpp.obj
[  7%] Building CXX object src/base/CMakeFiles/cvc5base.dir/listener.cpp.obj
[  7%] Building CXX object src/base/CMakeFiles/cvc5base.dir/output.cpp.obj
[  7%] Building CXX object src/base/CMakeFiles/cvc5base.dir/versioninfo.cpp.obj
[  7%] Built target cvc5base
[  7%] Building CXX object src/parser/CMakeFiles/cvc5parser-objs.dir/tokens.cpp.obj
[  7%] Built target cvc5parser-objs
[  7%] Building CXX object src/parser/CMakeFiles/cvc5parserapi-objs.dir/__/api/c/cvc5_parser.cpp.obj
[  7%] Building CXX object src/parser/CMakeFiles/cvc5parserapi-objs.dir/__/api/cpp/cvc5_parser.cpp.obj
[  7%] Built target cvc5parserapi-objs
CMake Error at D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/CoCoA-EP-stamp/CoCoA-EP-build-Production.cmake:37 (message):
  Command failed: 2

   'make' 'library'

  See also

    D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/CoCoA-EP-stamp/CoCoA-EP-build.log


CMake Error at D:/a/what4-solvers/what4-solvers/repos/cvc5/build/deps/src/CoCoA-EP-stamp/CoCoA-EP-build-Production.cmake:47 (message):
  Stopping after outputting logs.


-- Log output is:
make[3]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
Rebuilding unified header file for CoCoALib (version 0.99800)
Rebuilding dependencies in <CoCoALib>/src/AlgebraicCore/
Rebuilding dependencies in <CoCoALib>/src/AlgebraicCore/TmpFactorDir/
Rebuilding dependencies in <CoCoALib>/src/AlgebraicCore/TmpFactorDir/linalg/
Rebuilding dependencies in <CoCoALib>/src/AlgebraicCore/TmpFactorDir/multivariate/
Rebuilding dependencies in <CoCoALib>/src/AlgebraicCore/TmpHilbertDir/
Rebuilding dependencies in <CoCoALib>/src/server/
Rebuilding dependencies in <CoCoALib>/src/CoCoA-5/
********************
*** BOOST absent ***   ==> not rebuilding CoCoA-5 dependencies
********************
Rebuilding dependencies in <CoCoALib>/src/tests/
Compiling CoCoALib source subdirectory AlgebraicCore/
-----------------------------------------------------
Compiling time.o
Compiling assert.o
Compiling utils.o
Compiling utils-gmp.o
Compiling VectorOps.o
Compiling ProgressReporter.o
Compiling BuildInfo.o (with special preprocessor flags)
Compiling obsolescent.o
Compiling exception.o
Compiling error.o
Compiling MemPool.o
Compiling GlobalManager.o
Compiling SignalWatcher.o
SignalWatcher.C: In constructor 'CoCoA::SignalWatcher::SignalWatcher(int, void (*)(int))':
SignalWatcher.C:38:22: error: aggregate 'CoCoA::sigaction sa' has incomplete type and cannot be defined
   38 |     struct sigaction sa;
      |                      ^~
SignalWatcher.C:43:5: error: 'sigemptyset' was not declared in this scope
   43 |     sigemptyset(&sa.sa_mask);
      |     ^~~~~~~~~~~
SignalWatcher.C:44:19: error: 'SA_RESTART' was not declared in this scope
   44 |     sa.sa_flags = SA_RESTART; // Restart functions if interrupted by handler
      |                   ^~~~~~~~~~
SignalWatcher.C:46:37: error: invalid use of incomplete type 'struct CoCoA::sigaction'
   46 |     myPrevSigactionPtr = new struct sigaction;
      |                                     ^~~~~~~~~
In file included from SignalWatcher.C:18:
../../include/CoCoA/SignalWatcher.H:61:12: note: forward declaration of 'struct CoCoA::sigaction'
   61 |     struct sigaction* myPrevSigactionPtr;
      |            ^~~~~~~~~
SignalWatcher.C:47:47: error: invalid use of incomplete type 'struct CoCoA::sigaction'
   47 |     if (sigaction(sig, &sa, myPrevSigactionPtr) != 0)
      |                                               ^
../../include/CoCoA/SignalWatcher.H:61:12: note: forward declaration of 'struct CoCoA::sigaction'
   61 |     struct sigaction* myPrevSigactionPtr;
      |            ^~~~~~~~~
SignalWatcher.C:49:7: warning: possible problem detected in invocation of 'operator delete' [-Wdelete-incomplete]
   49 |       delete myPrevSigactionPtr;
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~
SignalWatcher.C:49:14: warning: invalid use of incomplete type 'struct CoCoA::sigaction'
   49 |       delete myPrevSigactionPtr;
      |              ^~~~~~~~~~~~~~~~~~
../../include/CoCoA/SignalWatcher.H:61:12: note: forward declaration of 'struct CoCoA::sigaction'
   61 |     struct sigaction* myPrevSigactionPtr;
      |            ^~~~~~~~~
SignalWatcher.C:49:7: note: neither the destructor nor the class-specific 'operator delete' will be called, even if they are declared when the class is defined
   49 |       delete myPrevSigactionPtr;
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~
SignalWatcher.C: In member function 'void CoCoA::SignalWatcher::myDeactivate()':
SignalWatcher.C:68:49: error: invalid use of incomplete type 'struct CoCoA::sigaction'
   68 |     sigaction(mySig, myPrevSigactionPtr, nullptr);
      |                                                 ^
../../include/CoCoA/SignalWatcher.H:61:12: note: forward declaration of 'struct CoCoA::sigaction'
   61 |     struct sigaction* myPrevSigactionPtr;
      |            ^~~~~~~~~
SignalWatcher.C:69:5: warning: possible problem detected in invocation of 'operator delete' [-Wdelete-incomplete]
   69 |     delete myPrevSigactionPtr;
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~
SignalWatcher.C:69:12: warning: invalid use of incomplete type 'struct CoCoA::sigaction'
   69 |     delete myPrevSigactionPtr;
      |            ^~~~~~~~~~~~~~~~~~
../../include/CoCoA/SignalWatcher.H:61:12: note: forward declaration of 'struct CoCoA::sigaction'
   61 |     struct sigaction* myPrevSigactionPtr;
      |            ^~~~~~~~~
SignalWatcher.C:69:5: note: neither the destructor nor the class-specific 'operator delete' will be called, even if they are declared when the class is defined
   69 |     delete myPrevSigactionPtr;
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~
make[5]: *** [../../configuration/autoconf.mk:97: SignalWatcher.o] Error 1
===============================================================================
>>>>>  Compilation failed in CoCoALib source subdirectory AlgebraicCore/  <<<<<
===============================================================================
make[4]: *** [Makefile:43: library] Error 1
make[3]: *** [Makefile:151: library] Error 2

make[2]: *** [CMakeFiles/CoCoA-EP.dir/build.make:87: deps/src/CoCoA-EP-stamp/CoCoA-EP-build] Error 1
make[1]: *** [CMakeFiles/Makefile2:845: CMakeFiles/CoCoA-EP.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```
</details>

**Diagnosis**

As noted in the `configure.sh` output, CVC5 is building against CoCoALib 0.99800. This is a somewhat old version of CoCoALib. If we look the first build error:

```
SignalWatcher.C: In constructor 'CoCoA::SignalWatcher::SignalWatcher(int, void (*)(int))':
SignalWatcher.C:38:22: error: aggregate 'CoCoA::sigaction sa' has incomplete type and cannot be defined
   38 |     struct sigaction sa;
      |                      ^~
```

This suggests that the build error occurs because the code is using Unix-like signal handling APIs that aren't available on Windows. That being said, this line number (`38`) has changed somewhat in the most recent CoCoALib release (0.99850), as seen [here](https://github.com/cocoa-official/CoCoALib/blob/c3c9bec71c96de263756e22048b66e1da105d2aa/src/AlgebraicCore/SignalWatcher.C#L38-L58). There is now a separate Windows-only code path (guarded by `#if defined (_WIN32) || defined (_WIN64)`) that wasn't present in CoCoALib 0.99800. (CoCoALib's `git` repo history only goes back to version 0.99855, so it's difficult to tell when exactly this Windows-only code path was introduced, but I would suspect that it was introduced some time between 0.99800 and 0.99850.)

Given this, I suspect that the issue would be resolved by upgrading from CoCoALib 0.99800 to 0.99850.

**Notes**

I reproduced this issue with CVC5 1.3.1. I could try repeating this with CVC5 1.3.4, but I doubt the results would change, as CVC5 1.3.4 also bundles the same version of CoCoALib.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -242,6 +242,27 @@ jobs:
             macos-target: 11.0
             gpl-tag: -gpl
 
+          # Windows builds with CoCoALib are generated for testing purposes only
+          # but are not currently uploaded.
+          - name: win64:production-gpl
+            os: windows-latest
+            config: production --auto-download --gpl --cocoa
+            cache-key: production-win64-native-gpl
+            strip-bin: strip
+            windows-build: true
+            shell: 'msys2 {0}'
+            gpl-tag: -gpl
+
+          - name: win64:production-arm64-gpl
+            os: windows-11-arm
+            config: production --auto-download --gpl --cocoa
+            cache-key: production-win64-native-arm64-gpl
+            strip-bin: strip
+            java-version: 21
+            windows-build: true
+            shell: 'msys2 {0}'
+            gpl-tag: -gpl
+
         exclude:
           - gpl-build: false
             build:
@@ -255,7 +276,12 @@ jobs:
           - gpl-build: false
             build:
               name: macos:production-arm64-gpl
-
+          - gpl-build: false
+            build:
+              name: win64:production-gpl
+          - gpl-build: false
+            build:
+              name: win64:production-arm64-gpl
 
     name: ${{ matrix.build.name }}
     runs-on: ${{ matrix.build.os }}

--- a/cmake/FindPoly.cmake
+++ b/cmake/FindPoly.cmake
@@ -15,8 +15,20 @@
 
 include(deps-helper)
 
+# On Windows we always link LibPoly statically, even into a shared libcvc5.
+# A LibPoly DLL auto-exports the GMP symbols it statically embeds (MinGW
+# auto-export), and those re-exported symbols (e.g. __gmp_default_allocate)
+# then collide with cvc5's own static GMP when linking libcvc5.dll, which lld
+# rejects with "<sym> was replaced". A static (PIC) LibPoly has no export
+# table, so there is nothing to collide.
+if(BUILD_SHARED_LIBS AND NOT WIN32)
+  set(POLY_BUILD_SHARED ON)
+else()
+  set(POLY_BUILD_SHARED OFF)
+endif()
+
 find_path(Poly_INCLUDE_DIR NAMES poly/poly.h)
-if(BUILD_SHARED_LIBS)
+if(POLY_BUILD_SHARED)
   find_library(Poly_LIBRARIES NAMES poly)
   find_library(PolyXX_LIBRARIES NAMES polyxx)
 else()
@@ -86,7 +98,7 @@ if(NOT Poly_FOUND_SYSTEM)
 
   set(Poly_INCLUDE_DIR "${DEPS_BASE}/include/")
 
-  if(BUILD_SHARED_LIBS)
+  if(POLY_BUILD_SHARED)
     set(POLY_BUILD_STATIC OFF)
     set(POLY_TARGETS poly polyxx)
     set(POLY_INSTALL_CMD
@@ -210,7 +222,7 @@ endif()
 set(Poly_FOUND TRUE)
 
 
-if(BUILD_SHARED_LIBS)
+if(POLY_BUILD_SHARED)
   add_library(Poly SHARED IMPORTED GLOBAL)
   add_library(Polyxx SHARED IMPORTED GLOBAL)
   if(CMAKE_SYSTEM_NAME STREQUAL "Windows")

